api/v1/create_run.go

```diff
@@ -0,0 +1,42 @@
+package v1
+
+import (
+	"github.com/SneaksAndData/nexus-core/pkg/checkpoint/models"
+	"github.com/SneaksAndData/nexus-core/pkg/checkpoint/request"
+	"github.com/SneaksAndData/nexus/services"
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"net/http"
+)
+
+func CreateRun(buffer *request.DefaultBuffer, configCache *services.MachineLearningAlgorithmCache) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		// TODO: log errors
+		algorithmName := ctx.Param("algorithmName")
+		payload := models.AlgorithmRequest{}
+		requestId := uuid.New()
+
+		if err := ctx.ShouldBindJSON(&payload); err != nil {
+			ctx.String(http.StatusBadRequest, `Algorithm payload is invalid: %s`, err.Error())
+			return
+		}
+
+		config, cacheErr := configCache.GetConfiguration(algorithmName)
+
+		if cacheErr != nil || config == nil {
```

Comment:
It would be better to log cache errors and 'not found in cache' events separately.